src/main/clojure/clara/rules/engine.cljc

```diff
@@ -72,14 +74,14 @@
 ;; Returns a new session with the additional facts inserted.
 (defprotocol ISession
 
-  ;; Inserts a fact.
-  (insert [session fact])
+  ;; Inserts facts.
+  (insert [session facts])
 
-  ;; Retracts a fact.
+  ;; Retracts facts.
   (retract [session fact])
 
   ;; Fires pending rules and returns a new session where they are in a fired state.
-  (fire-rules [session])
+  (fire-rules [session opts])
```

Comment:
My thinking is that this alters the ISession interface though. However I guess an implement of would have to implement both stories anyways due to no default behaviors able to be assigned in CLJ.